src/xiopak.cpp

```diff
@@ -1124,7 +1124,18 @@ SEC xfclear(	// Cleans the file by discarting all the data in the file
 	return xf->xflsterr;	/*    bad: report error */
 }
 //=============================================================================
-
+void xfjoinpath(			// Joins two directory path together
```

Comment:
I think xfjoinpath should take an additional arg which is the size of fullpath.  The code should use this to avoid overrunning the buffer.  Perhaps make the return type bool and return false if fullpath is truncated.

Unfortunately I do not see an easy way to determine of fullpath has been truncated.  Probably need a variant of strncpy0 that detects truncation.

Reply:
Maybe another case for moving towards more uniform adoption of `std::string`?

Reply:
Do I still return a truncated version of fullpath if False?